### PR TITLE
Fix conformance test flags are exported to other libraries that use the conformance test suite

### PR DESCRIPTION
**Pull request by hanxiaop**
**What type of PR is this?**

**What this PR does / why we need it**:

/kind bug
/area conformance

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-sigs/gateway-api/issues/2841.

Currently when implementing [GEP-2162](https://github.com/kubernetes-sigs/gateway-api/blob/f80f447fbf6e759a196cab6532b70582d9a8bd70/geps/gep-2162/index.md), the `features` need to be imported.  However, in the test suite, this import results in the variation of flags within our production code. The goal is to allow the importation of features without modifying the product code.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by hanxiaop** [comment]
@keithmattix @kflynn can you PTAL when you get a chance? Thanks!

**Comment by hanxiaop** [comment]
Also cc @LiorLieberman to take a look, the feature was introducted in https://github.com/istio/istio/pull/48770.

**Comment by LiorLieberman** [comment]
Thanks @hanxiaop! 
Can you elaborate on the problem? I am not sure I understand

**Comment by hanxiaop** [comment]
> Thanks @hanxiaop! Can you elaborate on the problem? I am not sure I understand

@LiorLieberman So basically, in that PR, you imported the test suite code into the product code:

https://github.com/istio/istio/blob/42572fab5be79af8de284953ec72cbf439631da8/pilot/pkg/config/kube/gateway/supported_features.go#L24,

and after https://github.com/kubernetes-sigs/gateway-api/pull/2801/files#diff-3532d0dde2d46785ab138cfe0fa7bddc5c1901dde4fe4079880451c010ea8254R34, the flag package was also imported into the suite. After that, our command-line tool was filled with gateway conformance test flags alongside our own pflags, since the test flags were imported in the dependency and thus in the flag set. This PR moves the features file out of the suite, so that we can directly import the features instead of the entire suite package and avoid issues like flags being altered.

**Comment by dprotaso** [comment]
cc @mlavacca - since you're refactoring the experimental suite

**Comment by howardjohn** [comment]
> Thanks, I think many implementations uses suite to import features. This would require all of them to move to features or their code wont work.



If we really want to maintain Go-level compatibility (which I think is a non-goal of the project, but I could be wrong) we can always add a type alias in `suite`.

File: hack/supportedfeatures/main.go [hunk 1]
```diff
@@ -21,14 +21,14 @@ import (
 	"sort"
 	"strings"
 
-	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/conformance/utils/features"
```
[the remark this answers is not in the document]
Comment (hanxiaop) [reply hack/supportedfeatures/main.go:24]: Great suggestion! Fixed.

**Comment by hanxiaop** [comment]
> Not sure whats the procedure in such case and whether it is considered breaking change or not. /cc @youngnick @shaneutt @robscott

I think this will not be an issue or a breaking change, as it's fairly common for bumping a dependency to result in a change in the package path.

**Review by mlavacca — COMMENTED** [review]
This looks good to me, @hanxiaop! Would you mind if we wait for https://github.com/kubernetes-sigs/gateway-api/pull/2868 to land before merging this one (that's part of the conformance suite epic and is required for 1.1)?

**Comment by hanxiaop** [comment]
> This looks good to me, @hanxiaop! Would you mind if we wait for #2868 to land before merging this one (that's part of the conformance suite epic and is required for 1.1)?

@mlavacca no problem, I'll rebase once it's merged, thanks for the review!

**Comment by hanxiaop** [comment]
@mlavacca PTAL, thanks!

**Comment by mlavacca** [comment]
> > > Thanks @hanxiaop! Can you elaborate on the problem? I am not sure I understand
> > 
> > 
> > @LiorLieberman So basically, in that PR, you imported the test suite code into the product code:
> > https://github.com/istio/istio/blob/42572fab5be79af8de284953ec72cbf439631da8/pilot/pkg/config/kube/gateway/supported_features.go#L24,
> > and after https://github.com/kubernetes-sigs/gateway-api/pull/2801/files#diff-3532d0dde2d46785ab138cfe0fa7bddc5c1901dde4fe4079880451c010ea8254R34, the flag package was also imported into the suite. After that, our command-line tool was filled with gateway conformance test flags alongside our own pflags, since the test flags were imported in the dependency and thus in the flag set. This PR moves the features file out of the suite, so that we can directly import the features instead of the entire suite package and avoid issues like flags being altered.
> 
> Thanks, I think many implementations uses `suite` to import features. This would require all of them to move to `features` or their code wont work.
> 
> Not sure whats the procedure in such case and whether it is considered breaking change or not. /cc @youngnick @shaneutt @robscott

Many changes have been made to the suite for its graduation; implementations will have to adapt to those changes in any case when bumping to the next Gateway API version. I do not see this kind of change as problematic.

**Review by mlavacca — CHANGES_REQUESTED** [review]
We need to change all the references to the features in the actual tests, under `conformance/tests`. As it is, the code does not even compile.

**Comment by hanxiaop** [comment]
> We need to change all the references to the features in the actual tests, under `conformance/tests`. As it is, the code does not even compile.

@mlavacca Fixed, missed that dependency. It would be great if someone could add the `ok-to-test` label so I can check if there are more things being broken.

**Comment by hanxiaop** [comment]
@mlavacca @youngnick @shaneutt @robscott @howardjohn @keithmattix PTAL, thanks!

**Review by howardjohn — APPROVED** [review]
/lgtm

can you rebase?

**Comment by hanxiaop** [comment]
@howardjohn PTAL again, thanks!

**Review by howardjohn — APPROVED** [review]
/lgtm

**Comment by hanxiaop** [comment]
@robscott PTAL, thanks!

**Comment by robscott** [comment]
Thanks @hanxiaop!

/approve

**Comment by robscott** [comment]
@hanxiaop if you'd prefer we can use #2980 which copied this PR and made some modifications to pass presubmits.

**Comment by hanxiaop** [comment]
> @hanxiaop if you'd prefer we can use #2980 which copied this PR and made some modifications to pass presubmits.

@robscott Thanks! I think it should be good now. New tests are continually being added to the folder, so I need to do a rebase again. If it doesn't work I think we can use that PR.

**Comment by hanxiaop** [comment]
@howardjohn just rebased and can you LGTM again? Thanks!

**Pull request merged**